routing/road_index.hpp

```diff
@@ -29,6 +29,17 @@ class RoadJointIds final
     return Joint::kInvalidId;
   }
 
+  Joint::Id GetLastJointId() const
+  {
+    Joint::Id lastJointId = Joint::kInvalidId;
+    for (Joint::Id const jointId : m_jointIds)
```

Comment:
Wow, wow! Linear complexity, not good! Polylog or constant is much better. Could you please just keep the last valid joint id and update this field on AddJoint.